### ISSUE TITLE
Mobile Popup

### ISSUE THREAD

**Issue by gregoryaroberts1** [cmv/cmv-app#455]
Hello, I am experiencing an issue when testing the CMV with a mobile device. PopupMobile only seems to work with the identify template(s) in the identify.js file located in the config folder. With an identify template (like the reverseGeocodeTemplate in the Geocoder widget), the mobile popup window only displays the title and no content.

![image](https://cloud.githubusercontent.com/assets/5385635/9550074/cec8a3ea-4d6e-11e5-8aa4-89722ddd8a55.png)

**Comment by tmcgee** [comment]
@gregoryaroberts1 Are you using your own modified reverseGeocodeTemplate or are you using the default one? If it is modified, can you please post it here in a comment.

**Comment by gregoryaroberts1** [comment]
I initially discovered this while developing a customized version of the Geocoder widget but I tested the demo site with a mobile device and am experiencing the same issue.

**Comment by tmcgee** [comment]
@gregoryaroberts1 yep. I have also verified this behavior with the default reverseGeocodeTemplate in the demo site. Looks like a bug to me. I've assigned it to myself for investigation and any fix to be included in the next cmv release.

**Comment by gregoryaroberts1** [comment]
Great! Thanks for looking into this.

**Comment by carrbrpoa** [comment]
I only manage to show something in mobile running reverse geocoder when I do 
```javascript
if (has('phone')) {
	this.map.infoWindow.setFeatures([graphic]);
}
```
in the end of Geocoder's `reverseGeocodeComplete` function.

**Comment by tmcgee** [comment]
@carrbrpoa Thanks for posting your findings. I haven't looked at the code but might that work without the IF statement for all browsers?

**Comment by carrbrpoa** [comment]
Actually it works! Just tested here. No IF needed.
But the results are different (with or without the IF). See:

Desktop/tablet:
![other](https://cloud.githubusercontent.com/assets/10212618/14217851/eba657d8-f826-11e5-9060-7678b2919d35.png)

Phone (2 steps needed to see the data):
![phone](https://cloud.githubusercontent.com/assets/10212618/14217864/f949a0ac-f826-11e5-836d-0838eeca4b7b.png)

**Comment by tmcgee** [comment]
Great to know!.

That two step process is the default way the ESRI mobile popup works.

**Comment by tmcgee** [comment]
Capturing a few items related to this for further discussion:

1. The ReveseGeocoding discussed above is part of the CMV Geocoder widget. The ESRI Geocoder widget has been deprecated and replaced a few versions back with a newer Search widget. The next release of CMV (develop branch) will support the Search widget. As a result we are removing support for CMV GeoCoder widget.

2. Reverse Geocoding uses the method `._task.locationToAddress` of the Geocoder widget. `_task` is not documented in the API. There likely is an equivalent object in the Search widget but we should move away from this. I propose we create a new ReverseGeocoder widget that would use [`esri/tasks/locator`](https://developers.arcgis.com/javascript/jsapi/locator-amd.html) which is documented and supported.

3. The right-click menu no longer works with v 3.15 and v 3.16 of the ESRI API. This is because ESRI changed the CSS class we use for a selector. In v 3.14 (and below), the selector was `.layersDiv`. In v 3.15 and v 3.16, the selector is now `.esriMapLayers`. This will be updated in the develop branch for the upcoming cmv release.

notifying @DavidSpriggs for opportunity to comment.

**Comment by tmcgee** [comment]
Original issue with Geocoder widget fixed in PR #531. Right click menu mentioned in last comment resolved in PR #530. Discussion on possible new ReverserGeocoder widget to be continued...
